### PR TITLE
[DC-1045] Remove snapshot-builder-manager

### PR DESCRIPTION
**Pull request by s-rubenstein**
Ticket: [<Link to Jira ticket>](https://broadworkbench.atlassian.net/browse/DC-1045)

What:

Removes snapshot builder manager role for favor of the steward role.

Why:

  \<For your reviewers' sake, please describe in ~1 paragraph what the value of this PR is to our users or to ourselves.\>

How:

  \<For your reviewers' sake, please describe in ~1 paragraph how this PR accomplishes its goal.\>

  \<If the PR is big, please indicate where a reviewer should start reading it (i.e. which file or function).\>

---

**PR checklist**

- [ ] I've followed [the instructions](https://github.com/broadinstitute/sam/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [ ] I've filled out the [Security Risk Assessment](https://sdarq.dsp-appsec.broadinstitute.org/jira-ticket-risk-assesment) (requires Broad Internal network access) and attached the result to the JIRA ticket

### CODE REVIEW

File: src/main/resources/reference.conf [hunk 1]
```diff
@@ -1311,6 +1311,9 @@ resourceTypes = {
       steward = {
```
Comment (evansuslovich) [inline src/main/resources/reference.conf:1308]: does the steward have the ability to update_snapshot_builder_settings?

Comment (s-rubenstein) [reply src/main/resources/reference.conf:1308]: We're gunna use `update_snapshot` as the action to check.

**Review by rjohanek — APPROVED** [review]

File: src/main/resources/reference.conf [hunk 2]
```diff
@@ -1327,12 +1330,6 @@ resourceTypes = {
       discoverer = {
         roleActions = ["discover_data", "read_policy::steward", "read_policy::discoverer", "read_auth_domain", "get_snapshot_builder_settings"]
       }
-      snapshot_builder_manager = {
-        roleActions = ["update_snapshot_builder_settings"]
```
Comment (snf2ye) [inline src/main/resources/reference.conf:1331]: Can we also remove the "update_snapshot_builder_settings" action?

Comment (snf2ye) [reply src/main/resources/reference.conf:1331]: Ah, I see your comments - Ok, my PR is blocking removing this.

**Review by snf2ye — APPROVED** [review]

File: src/main/resources/reference.conf (hunk 2 above)
Comment (s-rubenstein) [reply src/main/resources/reference.conf:1331]: Oh, I was going to say I think we can. Its `datasets` which still use `update_snapshot_builder_settings`

**Review by fboulnois — APPROVED** [review]
👍

**Pull request merged**